### ISSUE TITLE
MATLAB/Octave support?

### ISSUE THREAD

**Issue by krishnakumarg1984** [hkupty/iron.nvim#236]
Hi,

Does this support MATLAB/Octave?  I am forced to use MATLAB in a recent project, and was looking for a suitable neovim REPL plugin.

**Comment by agdestein** [comment]
You can add a MATLAB entry to the list of `repl_definition`s:
```
matlab = {
    command = {
        "matlab",
        "-nosplash",
        "-nodesktop",
    },
}
```

**Comment by krishnakumarg1984** [comment]
Yes. While that works, there doesn't seem to be support for sending a demarcated block of code (called cells in MATLAB parlance) to the REPL.

The last time I used MATLAB was about 5 years ago, and there was a vim script plugin (https://github.com/daeyun/vim-matlab) that worked reasonably well to provide a REPL with the ability to execute cells etc. This is unmaintained currently, and I am looking for a generic REPL plugin that will work for my python, julia, bash and matlab projects.

### Cell support (feature request)

Cells are code regions that begin with `%% ` and end when another cell starting block is encountered or the end of file is reached. The built-in MATLAB editor (although pretty horrible)  has support for "executing current cell" and "executing current cell and move to start of next" etc.

It would be nice if this plugin can provide something similar?

**Comment by hkupty** [comment]
Some repls support bracketed paste, which iron has implemented already. 
For this case, you can add the following to your repl definition and it should work:

```lua
matlab = {
    open = "%%",
    close = "%%",
    command = {
        "matlab",
        "-nosplash",
        "-nodesktop",
    },
}


```

**Comment by hkupty** [comment]
Or maybe I misunderstood what you said.

Since you can send motions to the repl, I would imagine this would be a feature better suited for a matlab specific plugin (to create a motion for a code block) then for iron. However, things might be a bit better once iron picks up support for treesitter.

I'm not really sure how to add this support to be fair, but it might be interesting to think about it even as a plugin on top of iron - that transforms literate code documents (documentation with code blocks) into executable code blocks using iron under the hood.

**Comment by v-i-s-h** [comment]
Using this function in `init.lua`

```lua
IronSendCell = function()
  local iron_core = require('iron.core')  
  local marks = require('iron.marks')
  
  local cell_pattern = "^\\s*##"  -- cell delimiter pattern
  local cell_start = vim.fn.search(cell_pattern, 'bcnW')
  local cell_end = vim.fn.search(cell_pattern, 'nW')

  local b_line = cell_start + 1
  local e_line = cell_end - 1
  local lines = vim.api.nvim_buf_get_lines(0, b_line - 1, e_line, 0)
  -- ignore blank lines
  local b_lines = {}
  for _, line in ipairs(lines) do
    if line:gsub("^%s*(.-)%s*$", "%1") ~= '' then
      table.insert(b_lines, line)
    end
  end
  if #b_lines == 0 then return end

  marks.set{
    from_line = b_line - 1,
    from_col = 0,
    to_line = e_line - 1,
    to_col = -1
  }
  marks.winrestview()
  
  iron_core.send(nil, b_lines)
  vim.fn.cursor(cell_end, 0) -- move to next cell start
end
```

Finally, map this to `Alt + Enter` using
```lua
vim.keymap.set('', '<M-Enter>', IronSendCell)
```

This will send the cell delimited by a pair of `##` to Iron REPL and move to next cell. Modify `cell_pattern` to suite to your need.
